site/gatsby-site/cypress/integration/wordcounts.js

```diff
@@ -0,0 +1,47 @@
+describe('The Word Counts Page', () => {
+  it('successfully loads', () => {
+    cy.visit('/summaries/wordcounts');
+  });
+
+  it('word count table should exist', () => {
+    cy.visit('/summaries/wordcounts');
+    cy.get('.row .list-group');
+    cy.get('.row .list-group > div');
+    cy.get('.row .list-group > div > span');
+  });
+
+  it('words should have length > 2 and count > 10', () => {
+    cy.visit('/summaries/wordcounts');
+
+    // Pull out the words and their counts and then check
+    // that the minimum word length and occurence count are accurate
+    let counts = [];
+
+    let words = [];
+
+    cy.get('.row .list-group > div')
+      .each((row) => {
+        words.push(row.text().trim());
+        let count = 0;
+
+        let countText = row.find('span').text().trim();
+
+        let spanIsNumeric = /\d+/.test(countText);
+
+        if (spanIsNumeric) {
+          count = parseInt(countText);
+        }
+        counts.push(count);
+      })
+      .then(() => {
+        const ascending = (a, b) => a - b;
+
+        expect(words.map((word) => word.length).sort(ascending)[0] > 2).to.be.true;
+        expect(counts.sort(ascending)[0] > 9).to.be.true;
+      });
+  });
+  it('word cloud should exist', () => {
+    cy.get('.col-8 svg');
```

Comment:
These CSS selectors are likely an indication that the word clouds should have a handle that can be checked for that is not related to the templating. Maybe a CSS class that has no associated styling (at present) would be good to add?

Reply:
Yes, cypress recommends using the `data-cy="something"` attribute to keep tests separate from CSS. It is not expected to add this to every component, but only the top ones.

https://docs.cypress.io/guides/references/best-practices#Selecting-Elements